File: src/main/java/space/maxkonkin/nasapicbot/config/YandexTranslateApiConfig.java
package space.maxkonkin.nasapicbot.config;


import lombok.AccessLevel;
import lombok.Getter;
import lombok.experimental.FieldDefaults;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;

@Configuration
@Getter
@FieldDefaults(level = AccessLevel.PRIVATE)
@PropertySource(value = "classpath:application-${SPRING_PROFILE}.yaml", factory = YamlPropertySourceFactory.class)
public class YandexTranslateApiConfig {
    @Value("${yandex-api.api-url}")
    String API_BASE_URI;
    @Value("${yandex-api.folder-id}")
    String FOLDER_ID;
    @Value("${yandex-api.api-token}")
    String API_TOKEN;
}

File: src/main/kotlin/space/maxkonkin/nasapicbot/config/YandexTranslateApiConfig.kt
package space.maxkonkin.nasapicbot.config

fun loadYandexTranslateConfig(properties: Map<String, Any>): YandexApiConfig {
    return YandexApiConfig(
        apiBaseUri = requireNotNull((properties["yandex-api"] as Map<*, *>)["api-url"]) as String,
        folderId = requireNotNull((properties["yandex-api"] as Map<*, *>)["folder-id"]) as String,
        apiToken = System.getenv("YA_API_TOKEN"),
        triggerApiUri = requireNotNull((properties["yandex-api"] as Map<*, *>)["trigger-api-url"]) as String,
        serviceAccountId = requireNotNull((properties["yandex-api"] as Map<*, *>)["service-account-id"]) as String,
        triggerFunctionId = requireNotNull((properties["yandex-api"] as Map<*, *>)["trigger-function-id"]) as String,
    )
}

data class YandexApiConfig(
    val apiBaseUri: String,
    val folderId: String,
    val apiToken: String,
    val triggerApiUri: String,
    val serviceAccountId: String,
    val triggerFunctionId: String,
)